### ISSUE TITLE
Follow user wrong indent?

### ISSUE THREAD

**Issue by sionking** [InstaPy/InstaPy#2888]
New follow_user function:
```
def follow_user(browser, track, login, user_name, button, blacklist, logger, logfolder):
    """ Follows the user from either its 'profile' page, a 'post' page or the users 'dialog' box """
    # check action availability
    if quota_supervisor("follows") == "jump":
        return False, "jumped"

    # available tracks are to follow in `profile`, `post` and `dialog`
    if track in ["profile", "post"]:
        if track == "profile":
            # check URL of the webpage, if it already is user's profile page, then do not navigate to it again
            user_link = "https://www.instagram.com/{}/".format(user_name)
            web_address_navigator(browser, user_link)

        try:
            sleep(2)
            follow_xpath = "//button[text()='Follow' or text()='Follow Back']"
            follow_button = browser.find_element_by_xpath(follow_xpath)

            if follow_button.is_displayed():
                click_element(browser, follow_button)

            else:
                browser.execute_script("arguments[0].style.visibility = 'visible'; "
                                       "arguments[0].style.height = '10px'; "
                                       "arguments[0].style.width = '10px'; "
                                       "arguments[0].style.opacity = 1",
                                            follow_button)

                click_element(browser, follow_button)

                # verify the last follow
                following, follow_button = get_following_status(browser,
                                                                 user_name,
                                                                  logger)
                if not following:
                    browser.execute_script("location.reload()")
                    sleep(2)
                    following, follow_button = get_following_status(browser,
                                                                     user_name,
                                                                      logger)
                    if following is None:
                        sirens_wailing, emergency_state = emergency_exit(browser,
                                                                          user_name,
                                                                           logger)
                        if sirens_wailing == True:
                            logger.warning("There is a serious issue: '{}'!\n".format(emergency_state))
                            return False, emergency_state

                        else:
                            logger.error("Unexpected failure happened after last follow!\n")
                            return False, "unexpected failure"

                    if following == False:
                        logger.warning("Last follow is not verified!\t~smells of a shadow ban\n")
                        sleep(600)
                        return False, "shadow ban"

                    else:
                        logger.info("Last follow is verified after reloading the page!\n")

        except NoSuchElementException:
            logger.info("--> '{}' is already followed".format(user_name))
            sleep(1)

            return False, "already followed"

        except StaleElementReferenceException:
            # https://stackoverflow.com/questions/16166261/selenium-webdriver-how-to-resolve-stale-element-reference-exception
            # 1. An element that is found on a web page referenced as a WebElement in WebDriver then the DOM changes
            # (probably due to JavaScript functions) that WebElement goes stale.
            # 2. The element has been deleted entirely.
            logger.error('--> element that is found on a web page referenced while the DOM changes')
            sleep(1)

            return False, "stale element"
```

```
# verify the last follow
....
```

Should be indent one less no?
This is when we don't do enough testing before committing.

@uluQulu

**Comment by uluQulu** [comment]
@sionking 
You have started to analyze code?? 😉
Good, you're right about indentation.

I have just applied it last times after your merged PR and what missing in there is the verification at `follow_button.is_displayed()` condition.
I also think it is not an important missing step cos earlier there were no any verification at all 😂

I will open a PR for it, thanks for pointing out.

--------

Cheers 😁
-------

**Comment by sionking** [comment]
@uluQulu 
I am now merging all of your stuff and of-course doing some QA (I can't have hing wrong here).
You code good, but complex, hard to follow, not really my style.

But it is only me you and tim right ?
